addons/mail/controllers/attachment.py

```diff
@@ -65,14 +65,11 @@ def mail_attachment_upload(self, ufile, thread_id, thread_model, is_pending=Fals
                     "res_model": "mail.compose.message",
                 }
             )
-        if request.env.user.share:
-            # Only generate the access token if absolutely necessary (= not for internal user).
-            vals["access_token"] = request.env["ir.attachment"]._generate_access_token()
         try:
             # sudo: ir.attachment - posting a new attachment on an accessible thread
             attachment = request.env["ir.attachment"].sudo().create(vals)
             attachment._post_add_create(**kwargs)
-            res = {"data": Store(attachment, extra_fields="access_token").get_result()}
+            res = {"data": Store(attachment).get_result()}
```

Comment:
I'm guessing this would be the place where the "delete" specific token that I'm talking about should be returned